extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/stork/StorkObservationCollectorBean.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.micrometer.runtime.binder.stork;
+
+import java.util.List;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Typed;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.smallrye.stork.api.ServiceInstance;
+import io.smallrye.stork.api.observability.ObservationCollector;
+import io.smallrye.stork.api.observability.ObservationPoints;
+
+@ApplicationScoped
+@Typed(ObservationCollector.class)
+public class StorkObservationCollectorBean implements ObservationCollector {
+
+    final MeterRegistry registry = Metrics.globalRegistry;
+
+    private final EventCompletionHandler STORK_HANDLER = ev -> {
+        //TODO
+    };
+    public static ObservationPoints.StorkResolutionEvent STORK_METRICS;
```

Comment:
I have doubts about this static var. Could we get away with a singleton producing a bean?

Reply:
ok, I will take a look next week.

Reply:
Yes, it will create one instance per Stork service. Do it's not right. 
You cannot use a singleton, you need a (concurrent) map service -> stork resolution event, and use `computeIfAbsent()`

Reply:
If believe we need tests with multiple services to see that this behavior is not correct.